src/pretix/base/services/invoices.py

```diff
@@ -71,16 +81,11 @@ def regenerate_invoice(invoice: Invoice):
             invoice.invoice_to = ""
 
         invoice.file = None
+        invoice.lines.all().delete()
```

Comment:
I haven't tested yet, but in the case of a new invoice this is called before the invoice even is created and has a primary key… does that actually work?